cfgov/v1/admin_views.py

```diff
@@ -17,14 +22,35 @@ def cdn_is_configured():
 
 
 def purge(url=None):
+    akamai_config = settings.WAGTAILFRONTENDCACHE.get('akamai', {})
+    cloudfront_config = settings.WAGTAILFRONTENDCACHE.get(
+        'cloudfront', {})
+
     if url:
-        purge_url_from_cache(url)
-        return "flushed %s" % url
+        # Use the Wagtail frontendcache PurgeBatch to perform the purge
+        batch = PurgeBatch()
+        batch.add_url(url)
+
+        # If the URL matches any of our cloudfront distributions, invalidate
+        # with that backend
+        if any(k for k in cloudfront_config.get('DISTRIBUTION_ID', {})
+               if k in url):
+            logger.info('Purging {} from cloudfront'.format(url))
```

Comment:
```suggestion
            logger.info('Purging {} from CloudFront'.format(url))
```

Reply:
These logging statements were actually intended to reference the `settings.WAGTAILFRONTENDCACHE` names. I've clarified the language on that somewhat.